extensions/README.md

```diff
@@ -0,0 +1,373 @@
+   than types. This implementation provides registration functions for new
+   values for those types.
+
+> [!NOTE]
+> Some of the definitions in the CoRIM spec are "imported" from the CoSWID
+> spec.  Some of those definitions also feature extension sockets. However, as
+> they are defined in a different spec (and are not implemented in this
+> library), they cannot be extended using the above mechanisms.
+
+## Map Extensions
+
+Map extensions allow extending CoRIM maps with additional keys, effectively
+defining new fields for the corresponding structures. In the code base, these
+can be identified by the embedded `Extensions` struct. These are
+
+- `comid.Comid`
+- `comid.Entity`
+- `comid.FlagsMap`
+- `comid.Mval`
+- `comid.Triples`
+- `corim.Entity`
+- `corim.Signer`
+- `corim.UnsignedCorim`
+
+To extend the above types, you need to define a struct containing your
+extensions and pass a pointer to an instance of that struct to the
+`RegisterExtensions()` method of the corresponding instance of the type that is
+being extended. This should be done as early as possible, before any marshaling
+is performed.
+
+These types can be extended in two ways: by adding additional fields, and by
+introducing additional constraints over existing fields.
+
+### Adding new fields
+
+To add a new fields, simply add them to your extensions struct, ensuring that
+the `cbor` and `json` tags on those fields are set corrects. As CoRIM spec
+mandates integer keys, you must use `keyasint` option for the `cbor` tag.
+
+To access the values of those fields, you can call the extended type instance's
+`Extensions.Get()` passing in the name of the field you want to access. The
+name can be either the Go struct's field name, the name specified in the `json`
+tag, or (a string containing) the integer specified in the `cbor` tag.
+
+`Get()` returns an `interface{}`. There are equivalent `GetInt()`,
+`GetString()`, etc. methods that  perform the required conversions, and return
+the value of the indicated type, along with possible errors. ("Must" versions
+of these also exists, e.g. `MustGetString()`, that do not return an error).
+
+You can also get the pointer your extensions instance itself by calling
+extended type instance's `GetExtensions()`. This returns an `interface{}`, so
+you will need to type assert in order to be able to access the fields directly.
+
+### Introducing additional constraints
+
+To introduce new constraints,  add a method called `Validate<type>(v *<type>)`
+to your extensions struct, where `<type>` is the name of the type being
+extended (one of the ones listed above) -- e.g.
+`ValidateComid(v *comid.Comid)` when extending `comid.Comid`. This method, if
+it exists, will be invoked inside the extended type instance's `Valid()`
+method, passing itself as the parameter.
+
+You do not need to define this method, unless you actually want to enforce some
+constraints (i.e. if you just want to define additional fields).
```

Comment:
```suggestion
constraints.
```

Reply:
Why remove this clarification?

Reply:
The text is confusing, are you providing additional information in the braces `( )` ?

I found it to be counter intuitive to your original statement, which says !
Perhaps may be we simply say:
You do not need to define this method, if  additional constraints are not introduced by the new fields.

Reply:
>if additional constraints are not introduced by the new fields.

??? the two things are orthogonal:

- new fields (do not require validation method -- constraints on new fields can be enforced via the fields' types)
- additional constraints (either on existing fields, or inter-field constraints)

The parentheciesd statement clarifies the situations where you might not want to add additional contraitas (when you're only adding new fields).

Reply:
Thanks for all the descriptions. I think, a clarification is needed at the top to describe what this extension implies.

Then this explaination can have the right context